rust/relay/src/server.rs

```diff
@@ -834,20 +868,25 @@ impl_stun_request_for!(Refresh, REFRESH);
 
 /// Private helper trait to make [`Server::verify_auth`] more ergonomic to use.
 trait ProtectedRequest {
-    fn message_integrity(&self) -> &MessageIntegrity;
-    fn username(&self) -> &Username;
+    fn message_integrity(&self) -> Option<&MessageIntegrity>;
```

Comment:
I think it's better if we use `Result` in here. We can use the same error type as you used in `verify_auth` buteven with an empty error type having a type marked as `must_use`  minimizes the chances of messing up in the future.